numpy/lib/npyio.py

```diff
@@ -599,13 +611,41 @@ def savez_compressed(file, *args, **kwds):
     """
     _savez(file, args, kwds, True)
 
+@contextlib.contextmanager
```

Comment:
Maybe something simpler?
```
@contextmanager
def TempFileName(suffix='', prefix='tmp', dir=None):
    fid, name = mkstemp(suffix, prefix, dir)
    os.close(fid)
    yield name
    os.remove(name)
```
This is a common pattern in a lot of tests, e.g, in `test_io.py`